src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -20,20 +23,30 @@ namespace Temporalio.Worker
     /// <summary>
     /// Instance of a workflow execution.
     /// </summary>
-    internal class WorkflowInstance : TaskScheduler, IWorkflowInstance
+    internal class WorkflowInstance : TaskScheduler, IWorkflowInstance, IWorkflowContext
     {
         private readonly TaskFactory taskFactory;
-        private readonly WorkflowInstanceDetails details;
+        private readonly WorkflowDefinition defn;
+        private readonly IPayloadConverter payloadConverter;
+        private readonly IFailureConverter failureConverter;
+        private readonly Lazy<WorkflowInboundInterceptor> inbound;
+        private readonly Lazy<WorkflowOutboundInterceptor> outbound;
```

Comment:
Do these need to be lazy?
You'll always have interceptors AFAICT.

Reply:
I should probably document this better, but it is important for me not to run all user code _inside_ our scheduler task. The constructor of this class is not run inside there. By properly waiting to instantiate the workflow class and properly waiting to build/init interceptors until we are inside the first job processed, I can ensure that `Workflow.Info` will work in workflow constructors and interceptors. If I eagerly instantiated either of those now you couldn't get workflow info.

Reply:
Yeah, when I get around to my full interceptor testing, it will include confirming workflow info is available in all places (and for workflow class constructor for that matter)